pkg/oc/cli/cli.go

```diff
@@ -330,14 +330,14 @@ func CommandFor(basename string) *cobra.Command {
 
 	switch basename {
 	case "kubectl":
-		cmd = kubecmd.NewKubectlCommand(in, out, errout)
+		cmd = kubecmd.NewDefaultKubectlCommand()
 	case "openshift-deploy":
 		cmd = deployer.NewCommandDeployer(basename)
 	case "openshift-recycle":
 		cmd = recycle.NewCommandRecycle(basename, out)
 	default:
 		shimKubectlForOc()
-		cmd = NewCommandCLI("oc", "oc", in, out, errout)
+		cmd = withPluginsSupport(NewCommandCLI("oc", "oc", in, out, errout), os.Args, out, errout)
```

Comment:
Make a wrapper similar to `NewDefaultKubectlCommandWithArgs` and use it here without the current nasty function in function.